ftdc/ftdc.go

```diff
@@ -183,6 +191,14 @@ func (ftdc *FTDC) Remove(name string) {
 func (ftdc *FTDC) Start() {
 	ftdc.readStatsWorker = utils.NewStoppableWorkerWithTicker(time.Second, ftdc.statsReader)
 	utils.PanicCapturingGo(ftdc.statsWriter)
+
+	// The `fileDeleter` goroutine mostly aligns with the "stoppable worker with ticker"
+	// pattern. But it has the additional desire that if file deletion exits with a panic, all of
+	// FTDC should stop.
+	utils.PanicCapturingGoWithCallback(ftdc.fileDeleter, func(err any) {
```

Comment:
When would `fileDeleter` panic?

Reply:
nobody knows!

Reply:
If we aren't aware of a `panic` path, then why introduce this code? The new path to `StopAndJoin` seems almost as worrying to me as FTDC causing a `viam-server` crash.

Reply:
We use them in all kinds of places that don't actually panic. I'm torn on that decision. But most relevant, the stoppable workers is in a panic capturing go and the stats writer are in one. So if this were not in one, that would raise more suspicion. Maybe?

Reply:
Discussed offline: what you have sounds good. The precedent does seem to be panic capturing.